sstables/sstables.cc

```diff
@@ -1595,6 +1598,21 @@ sstable::write_scylla_metadata(const io_priority_class& pc, shard_id shard, ssta
         _components->scylla_metadata->data.set<scylla_metadata_type::SSTableOrigin>(std::move(o));
     }
 
+    static thread_local struct {
+        bytes version;
+        bytes build_id;
+    } version_cache;
+    if (version_cache.version.empty()) {
+        version_cache.version = bytes(to_bytes_view(sstring_view(scylla_version())));
+        version_cache.build_id = bytes(to_bytes_view(sstring_view(get_build_id())));
+    }
```

Comment:
Is this caching necessary? How much work is it to get these two fields? Is it significant compared to writing an sstable?

Reply:
The work involved in getting the build id is not big, I'm just not sure that it'd be necessary to keep its containing executable page mapped in RAM at all otherwise.  But in any case the caching logic (if any) should be in `get_build_id()` itself.
(Caching Scylla version is totally pointless by itself, I added it for consistency)

Reply:
I agree. I think the best way forward is to remove the caching from this code and add it to `get_build_id()`, either in this PR or in a follow-up (I'm fine with either).